pkg/kritis/cryptolib/verifier.go

```diff
@@ -42,20 +46,67 @@ type PublicKey struct {
 	// KeyData holds the raw key material which can verify a signature.
 	KeyData []byte
 	// ID uniquely identifies this public key. For PGP, this should be the
-	// OpenPGP RFC4880 V4 fingerprint of the key.
+	// OpenPGP RFC4880 V4 fingerprint of the key. For PKIX and JWT, this should
+	// be a StringOrURI: it must either not contain ":" or be a valid URI.
 	ID string
 }
 
 // NewPublicKey creates a new PublicKey. `keyType` contains the type of the
 // public key, one of Pgp, Pkix or Jwt. `keyData` contains the raw key
 // material. `keyID` contains a unique identifier for the public key. For PGP,
-// this should be the OpenPGP RFC4880 V4 fingerprint of the key.
-func NewPublicKey(keyType KeyType, keyData []byte, keyID string) PublicKey {
-	return PublicKey{
+// this field should be left blank. The ID will be the OpenPGP RFC4880 V4
+// fingerprint of the key. For PKIX and JWT, this may be left blank, and the ID
+// will be generated based on the DER encoding of the key. If not blank, the ID
+// should be a StringOrURI: it must either not contain ":" or be a valid URI.
+func NewPublicKey(keyType KeyType, keyData []byte, keyID string) (*PublicKey, error) {
```

Comment:
Just curious, why return `*PublicKey` instead of `PublicKey`?

Reply:
Following Go convention where constructor named `NewThing()` returns `*Thing`.

Reply:
Gotcha. Thanks!